go/vt/wrangler/vexec.go

```diff
@@ -739,8 +737,27 @@ func (wr *Wrangler) getStreams(ctx context.Context, workflow, keyspace string) (
 		if len(nqr.Rows) == 0 {
 			continue
 		}
+		// Get all copy states for the shard.
+		var vreplIDs []int64
+		for _, row := range nqr.Rows {
+			vreplID, err := row.ToInt64("id")
+			if err != nil {
+				return nil, err
+			}
+			vreplIDs = append(vreplIDs, vreplID)
+		}
+		copyStatesByVReplID, err := wr.getCopyStates(ctx, primary, vreplIDs)
+		if err != nil {
+			return nil, err
+		}
 		for _, row := range nqr.Rows {
-			status, sk, err := wr.getReplicationStatusFromRow(ctx, row, primary)
+			vreplID, err := row.ToInt64("id")
+			if err != nil {
+				return nil, err
+			}
+
+			copyStates := copyStatesByVReplID[vreplID]
```

Comment:
We should check that a value exists, just to be safe, and return an error if not.

Reply:
Is that definitely an error condition? What if there is a `_vt.vreplication` row but copy state has not been created yet? In this case, `copyStates` will just be `nil`, which seems OK to me.